--- "CMakeLists.txt"	2023-12-06 03:21:12.000000000 +0800
+++ "CMakeLists.txt"	2024-03-10 20:21:45.779439965 +0800
@@ -80,22 +80,6 @@
     find_package(KF5 ${KF5_MIN_VERSION} REQUIRED COMPONENTS KDELibs4Support) # kcms/dateandtime
 endif()
 
-find_package(AccountsQt${QT_MAJOR_VERSION} ${ACCOUNTSQT_DEP_VERSION} CONFIG)
-
-set_package_properties(AccountsQt${QT_MAJOR_VERSION} PROPERTIES
-    DESCRIPTION "Accounts management library for Qt applications"
-    URL "https://gitlab.com/accounts-sso/libaccounts-qt"
-    PURPOSE "Required for building the OpenDesktop integration plugin"
-    TYPE OPTIONAL
-)
-
-find_package(KAccounts 20.04)
-set_package_properties(KAccounts PROPERTIES
-    DESCRIPTION "Accounts management library for KDE applications"
-    PURPOSE "Required for building the OpenDesktop integration plugin"
-    TYPE OPTIONAL
-)
-
 find_package(PkgConfig REQUIRED)
 pkg_check_modules(SignOnOAuth2 IMPORTED_TARGET signon-oauth2plugin)
 set_package_properties(signon-oauth2plugin PROPERTIES
@@ -131,17 +115,7 @@
     PURPOSE "Required for building Tablet input KCM"
 )
 
-find_package(LibKWorkspace ${PROJECT_VERSION} CONFIG REQUIRED)
-find_package(LibNotificationManager ${PROJECT_VERSION} CONFIG REQUIRED)
-find_package(LibTaskManager ${PROJECT_VERSION} CONFIG REQUIRED)
-find_package(LibColorCorrect ${PROJECT_VERSION} CONFIG REQUIRED)
-find_package(KWinDBusInterface CONFIG REQUIRED)
-find_package(ScreenSaverDBusInterface CONFIG REQUIRED)
-find_package(KRunnerAppDBusInterface CONFIG REQUIRED)
-find_package(KSMServerDBusInterface CONFIG REQUIRED)
 find_package(KF5ItemModels CONFIG REQUIRED)
-find_package(KSysGuard CONFIG REQUIRED)
-find_package(KDED CONFIG REQUIRED)
 
 find_package(KF5Baloo ${KF5_MIN_VERSION})
 set_package_properties(KF5Baloo PROPERTIES
@@ -238,24 +212,7 @@
 
 plasma_install_package(desktoppackage org.kde.plasma.desktop shells shell)
 
-add_subdirectory(layout-templates)
-
-add_subdirectory(doc)
-add_subdirectory(runners)
-add_subdirectory(containments)
-add_subdirectory(toolboxes)
-add_subdirectory(applets)
-add_subdirectory(kcms)
-add_subdirectory(knetattach)
-if(KAccounts_FOUND AND AccountsQt${QT_MAJOR_VERSION}_FOUND)
-    add_subdirectory(attica-kde)
-endif()
-add_subdirectory(imports/activitymanager/)
-add_subdirectory(solid-device-automounter)
 add_subdirectory(emojier)
-if(X11_Xkb_FOUND AND XCB_XKB_FOUND)
-    add_subdirectory(kaccess)
-endif()
 install(FILES org.kde.plasmashell.metainfo.xml DESTINATION ${KDE_INSTALL_METAINFODIR})
 
 # add clang-format target for all our real source files
