diff --git a/src/surfaceitem_x11.cpp b/src/surfaceitem_x11.cpp
index 6de9a73479d7b2c59abe5ac1d04a5cd5178418d9..fef00eb73e9dde249401febd8248c1e56a472e27 100644
--- a/src/surfaceitem_x11.cpp
+++ b/src/surfaceitem_x11.cpp
@@ -54,7 +54,6 @@ bool SurfaceItemX11::fetchDamage()
     if (!m_isDamaged) {
         return false;
     }
-    m_isDamaged = false;
 
     if (m_damageHandle == XCB_NONE) {
         return true;
@@ -105,6 +104,7 @@ void SurfaceItemX11::waitForDamage()
     free(reply);
 
     addDamage(region);
+    m_isDamaged = false;
 }
 
 void SurfaceItemX11::destroyDamage()
